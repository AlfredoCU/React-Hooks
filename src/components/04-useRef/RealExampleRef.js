import React, { useState } from "react";
import Quotes from "../03-examples/Quotes";

const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h4>REAL EXAMPLE USEREF</h4>
      {show && <Quotes />}
      <button className="btn btn-primary" onClick={() => setShow(!show)}>
        Mostrar/Desmontar
      </button>
    </div>
  );
};

export default RealExampleRef;
