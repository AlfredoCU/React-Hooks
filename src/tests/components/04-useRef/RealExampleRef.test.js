import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe("Pruebas en el componente RealExampleRef", () => {
  const wrapper = shallow(<RealExampleRef />);

  test("No muestra el componente Quotes", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Quotes").exists()).toBe(false);
  });

  test("Muestra el componente Quotes", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Quotes").exists()).toBe(true);
  });
});
