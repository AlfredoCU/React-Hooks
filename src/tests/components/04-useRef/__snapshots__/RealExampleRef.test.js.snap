// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente RealExampleRef Muestra el componente Quotes 1`] = `
<div>
  <h4>
    REAL EXAMPLE USEREF
  </h4>
  <Quotes />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Mostrar/Desmontar
  </button>
</div>
`;

exports[`Pruebas en el componente RealExampleRef No muestra el componente Quotes 1`] = `
<div>
  <h4>
    REAL EXAMPLE USEREF
  </h4>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Mostrar/Desmontar
  </button>
</div>
`;
